lib/os/heap.c

```diff
@@ -9,6 +9,10 @@
 
 static void *chunk_mem(struct z_heap *h, chunkid_t c)
 {
+	if (c == 0) {
+		return NULL;
+	}
+
```

Comment:
is it OK to declare variable not in the beginning of the scope block?

Reply:
No one complains when I do it, so I haven't stopped. :)

I don't know that we have an official policy on this.  Linux kernel style definitely forbids it, but our checkpatch variant doesn't warn about it (I don't know if upstream does and we disabled it, or if we're stale, or whatever).

But I do it all the time.  IMHO locally-declared variables are a big win for readability and maintainability.